algoliasearch-client-js/algoliasearch-client-js.d.ts

```diff
@@ -0,0 +1,1563 @@
+     * 'ignorePlurals': alternative words added by the ignorePlurals feature
+     * 'singleWordSynonym': single-word synonym (For example "NY" = "NYC")
+     * 'multiWordsSynonym': multiple-words synonym
+     * https://github.com/algolia/algoliasearch-client-js#alternativesasexact
+     */
+    alternativesAsExact?:any;
+    /**
+     * If set to 1, enables the distinct feature, disabled by default, if the attributeForDistinct index setting is set.
+     * https://github.com/algolia/algoliasearch-client-js#distinct
+     */
+    distinct?:any;
+    /**
+     * If set to true, the result hits will contain ranking information in the _rankingInfo attribute.
+     * default: false
+     * https://github.com/algolia/algoliasearch-client-js#getrankinginfo
+     */
+    getRankingInfo?:boolean;
+    /**
+     * All numerical attributes are automatically indexed as numerical filters
+     * default: ''
+     * https://github.com/algolia/algoliasearch-client-js#numericattributestoindex
+     */
+    numericAttributesToIndex?:string[];
+    /**
+     * @deprecated please use filters instead
+     * A string that contains the comma separated list of numeric filters you want to apply.
+     * https://github.com/algolia/algoliasearch-client-js#numericfilters-deprecated
+     */
+    numericFilters?:string[];
+    /**
+     * @deprecated
+     * Filter the query by a set of tags.
+     * https://github.com/algolia/algoliasearch-client-js#tagfilters-deprecated
+     */
+    tagFilters?:string;
+    /**
+     * @deprecated
+     * Filter the query by a set of facets.
+     * https://github.com/algolia/algoliasearch-client-js#facetfilters-deprecated
+     */
+    facetFilters?:string;
+    /**
+     * If set to false, this query will not be taken into account in the analytics feature.
+     * default true
+     * https://github.com/algolia/algoliasearch-client-js#analytics
+     */
+    analytics?:boolean;
+    /**
+     * If set, tag your query with the specified identifiers
+     * default: null
+     * https://github.com/algolia/algoliasearch-client-js#analyticstags
+     */
+    analyticsTags?:string[];
+    /**
+     * If set to false, the search will not use the synonyms defined for the targeted index.
+     * default: true
+     * https://github.com/algolia/algoliasearch-client-js#synonyms
+     */
+    synonyms?:boolean;
+    /**
+     * If set to false, words matched via synonym expansion will not be replaced by the matched synonym in the highlighted result.
+     * default: true
+     * https://github.com/algolia/algoliasearch-client-js#replacesynonymsinhighlight
+     */
+    replaceSynonymsInHighlight?:boolean;
+    /**
+     * Configure the precision of the proximity ranking criterion
+     * default: 1
+     * https://github.com/algolia/algoliasearch-client-js#minproximity
+     */
+    minProximity?:number;
+}
+declare module "algoliasearch" {
+    var algoliasearch:(a:string, b:string, options? : IClientOptions) => IAlgoliaClient;
```

Comment:
Prefer a function declaration, and use descriptive parameter names (likely `applicationId` and `apiKey`).